lib/libc/picolibc/Kconfig

```diff
@@ -3,6 +3,14 @@
 
 if PICOLIBC
 
+config PICOLIBC_USE_MODULE
```

Comment:
And here's the dependency loop if we want to base `PICOLIBC_SUPPORTED` on whether the user is trying to use the picolibc module with a project using C++ -- `PICOLIBC` should depend on `PICOLIBC_SUPPORTED`, but `PICOLIBC_SUPPORTED` needs to depend on `!CPLUSPLUS || !PICOLIBC_USE_MODULE`.

Reply:
no, `PICOLIBC_SUPPORTED` only needs to depend on `!(zephyr==0.15 && c++)`.
as you mentioned, other toolchain may likely have support for picolibc, even when C++ is used.

The fact that `PICOLIBC_USE_MODULE` depends on `!C++` and `PICOLIBC` does not form a loop in Kconfig as it is in same straight forward chain of dependencies.

See also other comment.

Reply:
ah, I was just thinking about this incorrectly -- I was thinking that `PICOLIBC_SUPPORTED` would depend on the user not selecting the module when C++ was in use. However, you've flipped that over and made the `PICOLIBC_USE_MODULE`  depend on not using C++, which makes a lot more sense.